### PR TITLE
feat: shell completions & auto releasing

### PR DESCRIPTION
**Pull request by TD-Sky**
Now yazi has shell completion scripts. I package them into the released assets.

The publish actions will be triggered after you create a new tag. Here is a real action:

https://github.com/TD-Sky/yazi/actions/runs/6535605738

And here are the released assets:

https://github.com/TD-Sky/yazi/releases/tag/v0.1.5

### CODE REVIEW

File: app/Cargo.toml
Comment (uncenter) [inline app/Cargo.toml:1]: These changes are just formatting? Probably shouldn't be included in this PR.

Comment (TD-Sky) [reply app/Cargo.toml:1]: I commit it by mistake, sorry! The completion scripts should be generated under `app`  in old plan, so I changed `app/Cargo.toml`.

Comment (uncenter) [reply app/Cargo.toml:1]: No worries! I was just taking a look at the PR and it caught my eye. Nice work!

**Comment by TD-Sky** [comment]
> if the completions can be generated, I wonder if it's worth committing them ? i.e., they can get out of date

It is not difficult at all to update them.

**Comment by sxyazi** [comment]
Thank you!
